src/inc/til/unicode.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace til
+{
+    static constexpr bool is_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xF800) == 0xD800;
+    }
+
+    static constexpr bool is_leading_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xFC00) == 0xD800;
+    }
+
+    static constexpr bool is_trailing_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xFC00) == 0xDC00;
+    }
+
+    constexpr std::wstring_view utf16_next(std::wstring_view wstr) noexcept
+    {
+        auto it = wstr.begin();
+        const auto end = wstr.end();
+        auto p = L"\uFFFD";
```

Comment:
I would like to avoid that as the `UNICODE_REPLACEMENT` value is defined in a "conhost header" and I feel like the `til` headers should be self-contained. I'm going to add a redundant `UNICODE_REPLACEMENT` variable in this header, so that it at least has the same name and is easier to find via code search tools.